openedx/core/djangoapps/user_api/management/commands/populate_enterprise_learner_current_job.py

```diff
@@ -0,0 +1,150 @@
+
+        existing_user_profile = UserProfile.objects.get(user=existing_user)
+
+        return existing_user_profile
+
+    def _update_current_job_of_learner(self, user_profile, current_job):
+        """
+        Update the current job of learner in their extended profile.
+
+        Args:
+            user_profile (UserProfile): Extended profile of the learner.
+            current_job (number): Current job of the learner.
+        """
+        try:
+            meta = user_profile.get_meta()
+            meta['enterprise_learner_current_job'] = current_job
+            user_profile.set_meta(meta)
+            user_profile.save()
+        except Exception:
+            LOGGER.exception(f'Error while updating current job of {user_profile.user.username}.')
+            raise
+
+    def _set_current_job_in_learner_profile(self, username, current_job):
+        """
+        Set the current job of learner in their extended profile.
+
+        Args:
+            username (string): Learner username for which we want to set current job.
+            current_job (number): Current job of the learner.
+        """
+        try:
+            LOGGER.info('Setting current job for user: %s as %s.', username, current_job)
+            existing_user_profile = self._get_user_profile(username)
+            self._update_current_job_of_learner(existing_user_profile, current_job)
+
+        except Exception:  # pylint: disable=bare-except
+            LOGGER.exception('Error while setting current job in learner profile.')
+            raise
+
+    def _populate_learner_current_job(self, response):
+        """
+        Populate the current job of learners in extended learner profile.
+        """
+        for username_current_job in response:
+            username = username_current_job['username']
+            current_job = username_current_job['current_job']
+            self._set_current_job_in_learner_profile(
+                username=username,
+                current_job=current_job,
+            )
+
+    def handle(self, *args, **options):
+        """
+        Handle the command.
+
+        Args:
+            *args: Variable length argument list.
+            **options: Arbitrary keyword arguments.
+        """
+        try:
+            LOGGER.info('Populating current job of learners in their extended profiles.')
+            response = self._fetch_learner_job_data(
+                config={
+                    'oauth_host': options['oauth_host'],
+                    'client_id': options['client_id'],
+                    'client_secret': options['client_secret'],
+                }
+            )
+            self._populate_learner_current_job(response)
+            LOGGER.info('Successfully populated current job of learners.')
+        except Exception as exc:  # pylint: disable=broad-except
+            LOGGER.exception('Error while populating learner current job.')
```

Comment:
I am not sure if this log will serve any purpose? We are re-raising the exception below and it will print the whole stack trace anyway.